pkg/volume/util/fsquota/project.go

```diff
@@ -86,31 +86,45 @@ func openAndLockProjectFiles() (*os.File, *os.File, error) {
 		err = fmt.Errorf("unable to open %s: %v", projectsFile, err)
 		return nil, nil, err
 	}
+
 	fProjid, err := os.OpenFile(projidFile, os.O_RDONLY|os.O_CREATE, 0644)
-	if err == nil {
-		// Check once more, to ensure nothing got changed out from under us
-		if err := projFilesAreOK(); err == nil {
-			err = lockFile(fProjects)
-			if err == nil {
-				err = lockFile(fProjid)
-				if err == nil {
-					return fProjects, fProjid, nil
-				}
-				// Nothing useful we can do if we get an error here
-				err = fmt.Errorf("unable to lock %s: %v", projidFile, err)
-				unlockFile(fProjects)
-			} else {
-				err = fmt.Errorf("unable to lock %s: %v", projectsFile, err)
-			}
-		} else {
-			err = fmt.Errorf("system project files failed re-verification: %v", err)
-		}
-		fProjid.Close()
-	} else {
+	if err != nil {
```

Comment:
@jingxu97 I squashed the commit and killed your comment asking about restructuring the code. I was hesitant to tug at the string of nested if-statements, but I've updated the PR to make `openAndLockProjectFiles()` more readable with the returns close to the error conditions instead of trickling down to the very end of the function.